### PR TITLE
PrivPol stuff for #271.

### PR DESCRIPTION
**Pull request by stwalkerster**
@dqwiki Pass me the text, and I'll include it

@FastLizard4 moar codereview pls.

### CODE REVIEW

**Comment by FastLizard4** [comment]
> @FastLizard4 moar codereview pls.

no u

File: templates/privacy/internal.tpl [hunk 1]
```diff
@@ -0,0 +1 @@
+We internally steal all your cookies, and feed them to four speedy reptilians roaming around a distant quadrant of the galaxy trying to get home.
```
Comment (FastLizard4) [inline templates/privacy/internal.tpl]: im in ur galaxy, fuel scooping ur main sequence stars

**Comment by FastLizard4** [comment]
FYI @dqwiki, still waiting for you to provide the final text to Simon before I begin formal code review here.

**Comment by stwalkerster** [comment]
@FastLizard4 I don't see why you can't review what's there already. I'm highly unlikely to change anything beyond the text in those two files from here on.

**Comment by FastLizard4** [comment]
@stwalkerster Fair enough, I'll go ahead with that then

File: privacy.php [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+require_once 'includes/PdoDatabase.php';
+require_once 'includes/SmartyInit.php';
+
+// eurgh. yes, this is needed. The internal header does a logged-in user check, which is a database call...
+if (Offline::isOffline()) {
+	echo Offline::getOfflineMessage(true);
+	die();
+}
+
+if(isset($_GET['internal'])){
```
Comment (FastLizard4) [inline privacy.php]: Code style (space after if)

**Comment by stwalkerster** [comment]
The privpol reformat renders as such - @dqwiki please can you confirm this is the intended format?

privpol reformat

File: templates/privacy/public.tpl [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+<p>By using this project, you agree that the volunteer administrators of this project will have access to any data you
+    submit. This can include your IP address, your username/password combination for accounts created in Labs services,
+    and any other information that you send. The volunteer administrators of this project are bound by the Wikimedia
+    Labs Terms of Use, and are not allowed to share this information or use it in any non-approved way.</p>
+<p>Since access to this information is fundamental to the operation of Wikimedia Labs, these terms regarding use of your
+    data expressly override the Wikimedia Foundation's Privacy Policy as it relates to the use and access of your
+    personal information.</p>
+
+<h3>What data do we collect, and why?</h3>
+<p>This system records your IP Address and useragent data. This information may be processed overseas and your data will
```
Comment (FastLizard4) [inline templates/privacy/public.tpl]: May want to explain in simple terms what useragent data is (e.g., "information about what software you're using to access this website, and the operating system your computer runs").

File: templates/privacy/public.tpl [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    remain confidential. It is important to note that this information is provided by your internet browser to any
+    website you visit; it is not possible to confirm any specific person's identity with this information. We only use
+    this to prevent abuse of Wikipedia and ACC.
+    We also require your email address so that we can respond to you with questions and the result of your request.
+    By submitting an account request at ACC, you agree to provide this information and allow ACC volunteers to view it
+    for the explicit purpose of reviewing your account request on the English Wikipedia.</p>
+
+<h3>How is this data stored, and who can see it?</h3>
+<p>We store this data in a secure database, which is visible only to ACC developers, several of whom are identified to
+    the Wikimedia Foundation, just as Checkusers are required to do. In order to assist with reviewing your block, this
```
Comment (FastLizard4) [inline templates/privacy/public.tpl]: "Identified to the Wikimedia Foundation" and "Checkusers" seem out of place here; a new user requesting an account (hell, even many experienced users I imagine) are unlikely to understand the significance of these terms.  It's also worth noting that anyone with access to this data is identified to the WMF, since they are just the tool roots, and they would be identified by virtue of membership in ACC.

File: templates/privacy/public.tpl [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    information is provided to ACC volunteers as follows:</p>
+<ul>
+    <li>Status update emails are sent through the system, but to prevent abuse and to be able to correlate you emailing
+        us, your email is visible all volunteers.</li>
+    <li>Your useragent will only be visible to ACC Senior Developers and those reviewing the block that have access to
```
Comment (FastLizard4) [inline templates/privacy/public.tpl]: "Reviewing the block" -> "reviewing your account request"

File: templates/privacy/public.tpl [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+        the CheckUser tool on Wikipedia. CheckUsers need to see your useragent to help differentiate you from the person
```
Comment (FastLizard4) [inline templates/privacy/public.tpl]: Again, "Checkuser" is probably too technical/jargony for use here.  Something simplified would be better; perhaps, "trusted users tasked with identifying long-term-abuse patterns" (suggested wording, would probably need further improvement).

File: templates/privacy/public.tpl [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+        the block is intended for, and UTRS developers need to see this information to ensure the tool is working
```
Comment (FastLizard4) [inline templates/privacy/public.tpl]: "UTRS"

File: templates/privacy/public.tpl [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+        properly.</li>
+    <li>Your IP address will be visible to ACC volunteers only while processing your request. After your request is
+        closed or before it is verified, only tool administrators, Senior Developers, and those that have access to the
+        CheckUser tool on Wikipedia have access to it. This is to be able to prevent abuse of the tool and Wikipedia.</li>
+</ul>
+<p>In very limited cases your IP address and useragent may be stored to prevent additional abuse of ACC or Wikipedia.
+    Your useragent and IP address additionally be visible upon request from an approved member of the Wikimedia
+    Foundation. The use of this is rare. You agree that any data released under this is subject to the Privacy Policy of
+    the Wikimedia Foundation, and this privacy policy is nullified for the released data only.
```
Comment (FastLizard4) [inline templates/privacy/public.tpl]: The "this" in "this privacy policy is nullified" is somewhat ambiguous; suggest rewording as "the privacy policy of the Account Creation Assistance system is nullified"

File: templates/privacy/public.tpl [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+    At no point will your data be provided to a third party for any purpose; furthermore, this information will be
+    removed from our system no more than one week after your appeal is closed. This removal process is automated, so you
```
Comment (FastLizard4) [inline templates/privacy/public.tpl]: Checking the config, unless I've misread it (perhaps @stwalkerster can confirm for me?), our dataclear time is currently set to 15 days.  Also, "appeal" -> "account request"

File: templates/privacy/public.tpl [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+    don't need to worry about anyone forgetting to hit the "delete" button.
+    Finally, we are not the Wikimedia Foundation. We are a group of volunteers who have made our identities known to the
+    foundation.</p>
+
+<h3>What are your rights with regard to this information?</h3>
+<p>If you wish to see what information has been collected on you by this system, you may email the development team at
```
Comment (FastLizard4) [inline templates/privacy/public.tpl]: Strictly speaking, it's not the development team, but the administrative team

File: templates/privacy/public.tpl [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+    enwiki-acc-admins@googlegroups.com to request all information associated with your appeal.
```
Comment (FastLizard4) [inline templates/privacy/public.tpl]: Another instance of "appeal" -> "account request"

File: templates/privacy/public.tpl [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+    If you do not wish for this information to be collected by ACC, you may appeal via your talk page on Wikipedia.
+    Please note that if you appeal on Wikipedia, your IP address and useragent may be examined by any CheckUser, with
+    cause, in accordance with the Wikimedia Foundation Privacy Policy.
```
Comment (FastLizard4) [inline templates/privacy/public.tpl]: I think the entire above three lines are only relevant to UTRS and can be removed from here.

File: templates/privacy/public.tpl [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+    If you have already entered a account request at ACC and wish for your information to be deleted immediately, please
```
Comment (FastLizard4) [inline templates/privacy/public.tpl]: Grammar pedant: "a" -> "an" 😛

File: templates/privacy/public.tpl [hunk 14]
```diff
@@ -0,0 +1,64 @@
...
+    email the development team at enwiki-acc-admins@googlegroups.com to have your appeal deleted from the database.
+    Again, please note that this information will be automatically removed one week after your appeal is resolved.
```
Comment (FastLizard4) [inline templates/privacy/public.tpl]: 15 days by config, as noted above

**Comment by stwalkerster** [comment]
@FastLizard4 I could see the commit by ID on enwikipedia-acc/waca, so I went to the browse files option, then created a new branch on that commit. Once I'd done that I could git fetch it as normal, then I deleted the temp branch.

**Comment by FastLizard4** [comment]
@stwalkerster Huh.  I thought I tried that, but still couldn't find the commit.  Oh well ¯\_(ツ)_/¯

File: templates/privacy/public.tpl (hunk 6 above)
Comment (dqwiki) [inline templates/privacy/public.tpl]: This one I wish to leave in place. If need be we can link to the CU page on wikipedia, but it defines a specific group of people, and not people who can do x task. At worst, we could also directly link to the Special:ListUsers checkuser group.

File: templates/privacy/public.tpl (hunk 8 above)
Comment (dqwiki) [inline templates/privacy/public.tpl]: I could use "this document", i'll put that in for now. The purpose here is to self reference, and not make the appearance that another policy needs reading.

File: templates/privacy/public.tpl (hunk 10 above)
Comment (dqwiki) [inline templates/privacy/public.tpl]: With regular tool admins not being able to access each bit of data associated with the appeal, we don't want to open them to legal requests for information they don't have. Reworded to senior development team.

File: templates/privacy/public.tpl (hunk 12 above)
Comment (dqwiki) [inline templates/privacy/public.tpl]: That's not possible. This is part of required legal compliance under Regulation 2016/679 Article 17, and while not yet in effect, I'd rather us be ready and compliant ahead of time.

Comment (FastLizard4) [inline templates/privacy/public.tpl]: What is "Regulation 2016/679 Article 17", and why does it mandate us noting that one can appeal blocks on their Wikipedia user talk page even though ACC does not do anything with unblock requests?  At a minimum, I think this should be reworded to be more relevant to ACC.
